requirements/adjust_versions.py

```diff
@@ -20,6 +20,9 @@
 def find_latest(ver: str) -> Dict[str, str]:
     # drop all except semantic version
     ver = re.search(r'([\.\d]+)', ver).groups()[0]
+    # in case there remaining dot at the end - e.g "1.9.0.dev20210504"
+    ver = ver[:-1] if ver[-1] == '.' else ver
+    print(f"finding ecosystem versions for: {ver}")
```

Comment:
This could be replaced for

```suggestion
ver = LooseVersion("1.9.0.dev20210504").version[:3]
ver = '.'.join(str(v) for v in ver)
```

Reply:
loose version has a problem with the appendix after version like commit hash